x/observer/types/message_add_observer.go

```diff
@@ -43,5 +46,18 @@ func (msg *MsgAddObserver) ValidateBasic() error {
 	if err != nil {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
 	}
+	_, err = sdk.AccAddressFromBech32(msg.ObserverAddress)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid observer address (%s)", err)
+	}
+	_, err = sdk.AccAddressFromBech32(msg.ZetaclientGranteeAddress)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid zetaclient grantee address (%s)", err)
+	}
+	_, err = common.NewPubKey(msg.ZetaclientGranteePubkey)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidPubKey, "invalid zetaclient grantee pubkey (%s)", err)
+	}
+	// https://github.com/zeta-chain/node/issues/988
```

Comment:
Not sure I understand, if it should be from the same pair, why not derivating the address from the pubkey?

Reply:
answered above

Reply:
also updated the issue mentioned here to reflect the same

Reply:
https://github.com/zeta-chain/zeta-node/blob/e8f8182d18d0ba4d1a64285d521e21c997994d51/common/pubkey.go#L36-L46